core/comments/workspace_comment.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Workspace} from '../workspace.js';
+import {Size} from '../utils/size.js';
+import {Coordinate} from '../utils/coordinate.js';
+import * as idGenerator from '../utils/idgenerator.js';
+
+export class WorkspaceComment {
+  /** The unique identifier for this comment. */
+  public readonly id: string;
+
+  /** The text of the comment. */
+  private text = '';
+
+  /** The size of the comment in workspace units. */
+  private size = new Size(120, 100);
+
+  /** Whether the comment is collapsed or not. */
+  private collapsed = false;
+
+  /** Whether the comment is editable or not. */
+  private editable = true;
+
+  /** Whether the comment is movable or not. */
+  private movable = true;
+
+  /** Whether the comment is deletable or not. */
+  private deletable = true;
+
+  /** The location of the comment in workspace coordinates. */
+  private location = new Coordinate(0, 0);
+
+  /** Whether this comment has been disposed or not. */
+  private disposed = false;
+
+  /**
+   * Constructs the comment.
+   *
+   * @param workspace The workspace to construct the comment in.
+   * @param id An optional ID to give to the comment. If not provided, one will
+   *     be generated.
+   */
+  constructor(
+    protected readonly workspace: Workspace,
+    id?: string,
+  ) {
+    // TODO: Before merging, file issue to update getCommentById.
+    this.id = id && !workspace.getCommentById(id) ? id : idGenerator.genUid();
+
+    // TODO: Before merging, file issue to add top comment.
+    // workspace.addTopComment(this);
+
+    // TODO(7909): Fire events.
+  }
+
+  /** Sets the text of the comment. */
+  setText(text: string) {
+    this.text = text;
+  }
+
+  /** Returns the text of the comment. */
+  getText(): string {
+    return this.text;
+  }
+
+  /** Sets the comment's size in workspace units. */
+  setSize(size: Size) {
+    this.size = size;
+  }
+
+  /** Returns the comment's size in workspace units. */
+  getSize(): Size {
+    return this.size;
+  }
+
+  /** Sets whether the comment is collapsed or not. */
+  setCollapsed(collapsed: boolean) {
+    this.collapsed = collapsed;
+  }
+
+  /** Returns whether the comment is collapsed or not. */
+  isCollapssed(): boolean {
```

Comment:
typo

Reply:
This is one thing that's bad about auto complete. I type it wrong once and then I just tab-complete wrong for the rest of time :P